digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/RoadLinkChangeClient.scala

```diff
@@ -0,0 +1,168 @@
+      }
+  },
+  {
+    case trafficDirection: TrafficDirection =>
+      trafficDirection match {
+        case TrafficDirection.BothDirections => JInt(0)
+        case TrafficDirection.TowardsDigitizing => JInt(1)
+        case TrafficDirection.AgainstDigitizing => JInt(2)
+        case _ => JNull
+      }
+  }
+  ))
+
+  object GeometrySerializer extends CustomSerializer[List[Point]](_ => (
+    {
+      case JString(lineString) =>
+        lineStringToPoints(lineString)
+    },
+    {
+      case points: List[Point] =>
+        JObject(JField("geometry", JString(""))) // not implemented until reverse operation is needed
+    }
+  ))
+
+  implicit val formats = DefaultFormats + changeItemSerializer + RoadLinkChangeTypeSerializer + GeometrySerializer +
+    AdminClassSerializer + TrafficDirectionSerializer
+
+  case class RoadLinkInfo(linkId: String, linkLength: Double, geometry: List[Point], roadClass: Int, adminClass: AdministrativeClass, municipality: Int, trafficDirection: TrafficDirection)
+  case class ReplaceInfo(oldLinkId: String, newLinkId: String, oldFromMValue: Double, oldToMValue: Double, newFromMValue: Double, newToMValue: Double)
+  case class RoadLinkChange(changeType: RoadLinkChangeType, oldLink: Option[RoadLinkInfo], newLinks: Seq[RoadLinkInfo], replaceInfo: Seq[ReplaceInfo])
+
+  def fetchLatestSuccessfulUpdateDate(): DateTime = {
+    // placeholder value as long as fetching this date from db is possible
+    DateTime.parse("2022-05-10")
+  }
+
+  def listFilesAccordingToDates(since: DateTime, until: DateTime) = {
+    def isValidKey(key: String): Boolean = {
+      try {
+        val keyParts = key.replace(".json", "").split("_")
+        val keySince = DateTime.parse(keyParts.head)
+        val keyUntil = DateTime.parse(keyParts.last)
+        !(keySince.isBefore(since) || keyUntil.isAfter(until)) // get no changes before or after the requested period
+      } catch {
+        case illegalArgument: IllegalArgumentException =>
+          logger.error("Key provides no valid dates.")
+          false
+        case e: Exception =>
+          logger.error(e.getMessage)
+          false
+      }
+    }
+
+    val request = ListObjectsV2Request.builder().bucket(s3Bucket).build()
+    val objects = s3Client.listObjectsV2(request).contents().asScala.toList
+    objects.map(_.key()).filter(key => isValidKey(key))
+  }
+
+  def fetchChangeSetFromS3(filename: String) = {
+    val s3Object = s3Service.getObjectFromS3(s3Bucket, filename)
+    fromInputStream(s3Object).mkString
+  }
+
+
+  def convertToRoadLinkChange(changeJson: String) : Seq[RoadLinkChange] = {
+    val json = parseJson(changeJson)
+    try {
+      json.extract[Seq[RoadLinkChange]]
+    } catch {
+      case e =>
+        logger.error(e.getMessage)
+        Seq.empty[RoadLinkChange]
+    }
+  }
+
+  def getRoadLinkChanges(since: DateTime = fetchLatestSuccessfulUpdateDate(), until: DateTime = DateTime.now()): Seq[RoadLinkChange] = {
```

Comment:
```suggestion
  def getRoadLinkChanges(since: DateTime = DateTime.parse("2022-05-10"), until: DateTime = DateTime.now()): Seq[RoadLinkChange] = {
```

Reply:
Onko joku syy, miksi haluat tämän jo tänne? Tuosta kovakoodatusta päivästä on tarkoitus luopua, kun fetchLatestSuccessfulUpdateDate() pääsee hakemaan viimeisimmän onnistuneen samuutuksen päivän kannasta.

Reply:
Ei nyt niin väliä mut vähän turha metodi joka palauttaa vaan yhden arvon, taakka poistaa joko metodi tai default arvolla ei taida olla mitään eroa, vähemmän vaan koodia.